srtcore/packetfilter.h

```diff
@@ -210,4 +210,6 @@ bool CheckFilterCompat(SrtFilterConfig& w_agent, SrtFilterConfig peer);
 inline void PacketFilter::feedSource(CPacket& w_packet) { SRT_ASSERT(m_filter); return m_filter->feedSource((w_packet)); }
 inline SRT_ARQLevel PacketFilter::arqLevel() { SRT_ASSERT(m_filter); return m_filter->arqLevel(); }
 
+bool ParseFilterConfig(std::string s, SrtFilterConfig& out, PacketFilter::Factory** ppf);
+
```

Comment:
How about
```
bool ParseFilterConfig(std::string s, SrtFilterConfig& out, PacketFilter::Factory** ppf = NULL);
```
and remove the implementation of `bool ParseFilterConfig(std::string s, SrtFilterConfig& out);`

I'm a little worried about these functions without "srt" prefix in global namespace. In this case, `ParseFilterConfig` can be a static member function of `PacketFilter`.

Please pass std::string by const reference.

Reply:
None of the code is mine. I am just fixing the build.

Reply:
The problem is that the 3-parameter version requires access to `PacketFilter::Factory` type definition, while the access to `ParseFilterConfig` function is also required in `packetfilter_api.h` file, which cannot include the file with that definition.